core/src/main/java/com/graphhopper/util/PathSimplification.java

```diff
@@ -0,0 +1,195 @@
+        int[] offset = new int[toSimplify.size()];
+        int[] endIntervals = new int[toSimplify.size()];
+        // All start at 0
+        int[] startIntervals = new int[toSimplify.size()];
+
+        int nonConflictingStart;
+        int nonConflictingEnd;
+        int toSimplifyIndex;
+        int toShiftIndex;
+
+        boolean simplificationPossible;
+
+        while (!endReached) {
+            endIntervals = calculateEndIntervals(endIntervals, startIntervals, offset, toSimplify);
+
+            simplificationPossible = true;
+            nonConflictingStart = 0;
+            nonConflictingEnd = Integer.MAX_VALUE;
+            toSimplifyIndex = -1;
+            toShiftIndex = -1;
+
+            for (int i = 0; i < toSimplify.size(); i++) {
+                if (startIntervals[i] >= nonConflictingEnd || endIntervals[i] <= nonConflictingStart) {
+                    simplificationPossible = false;
+                }
+                if (startIntervals[i] > nonConflictingStart) {
+                    toSimplifyIndex = -1;
+                    nonConflictingStart = startIntervals[i];
+                }
+                if (endIntervals[i] < nonConflictingEnd) {
+                    toSimplifyIndex = -1;
+                    nonConflictingEnd = endIntervals[i];
+                    // Remember the lowest endInterval
+                    toShiftIndex = i;
+                }
+                if (startIntervals[i] >= nonConflictingStart && endIntervals[i] <= nonConflictingEnd) {
+                    toSimplifyIndex = i;
+                }
+            }
+
+            if (toSimplifyIndex >= 0 && simplificationPossible) {
+                // Only simplify if there is more than one point
+                if(nonConflictingEnd - nonConflictingStart > 1){
+                    // Simplify
+                    int removed = douglasPeucker.simplify(pointList, nonConflictingStart, nonConflictingEnd -1 );
+                    if (removed > 0) {
+                        for (int i = 0; i < toSimplify.size(); i++) {
+                            reduceNumberOfPoints(toSimplify.get(i), offset[i], removed, startIntervals[i], endIntervals[i] - removed);
+                        }
+                    }
+                }
+            }
+
+            if (toShiftIndex < 0) {
+                throw new IllegalStateException("Shift Index cannot be below 0");
+            }
+            int length = getNumberOfPoints(toSimplify.get(toShiftIndex), offset[toShiftIndex]);
+            startIntervals[toShiftIndex] += length;
+            offset[toShiftIndex]++;
+            if (offset[toShiftIndex] >= getLength(toSimplify.get(toShiftIndex))) {
+                endReached = true;
+            }
+        }
+        return pointList;
+    }
+
+    private int getLengthIL(InstructionList il) {
+        return il.size();
+    }
+
+    private int getLengthPD(PathDetails pd) {
+        return pd.getDetails().size();
+    }
+
+    private int getLength(Object o) {
```

Comment:
Yes I had the same idea. Not sure if it's worth the troubles. Should I try if it's possible?